tests/model_tests.cpp

```diff
@@ -669,4 +667,92 @@ TEST_CASE("null to JSON", "[model]") {
     char *json = Foo_to_json(nullptr, 0, nullptr);
 
     CHECK(json == nullptr);
+}
+
+#define LISTS_MODEL(xx, ...) \
+xx(fruit, Fruit, list, fruit, __VA_ARGS__)\
+xx(errors, string, list, errors, __VA_ARGS__)\
+xx(codes, int, list, codes, __VA_ARGS__)
+
+DECLARE_MODEL(ListsObj, LISTS_MODEL)
+
+IMPL_MODEL(ListsObj, LISTS_MODEL)
+
+TEST_CASE("parse model_list", "[model]") {
```

Comment:
not exactly relevant to this pr only but are there tests that 'junk' json doesn't cause a crash but instead returns an empty list? same question for extra json fields or is this rigid parsing only?